test/optimizer/operator_transformer_test.cpp

```diff
@@ -0,0 +1,838 @@
+    std::vector<catalog::Schema::Column> cols_a;
+    cols_a.emplace_back("a1", type::TypeId::INTEGER, true, int_default);
+    cols_a.emplace_back("a2", type::TypeId::VARCHAR, 20, true, varchar_default);
+    auto schema_a = catalog::Schema(cols_a);
+
+    table_a_oid_ = accessor_->CreateTable(accessor_->GetDefaultNamespace(), "a", schema_a);
+    auto table_a = new storage::SqlTable(&block_store_, schema_a);
+    EXPECT_TRUE(accessor_->SetTablePointer(table_a_oid_, table_a));
+
+    txn_manager_->Commit(txn_, TestCallbacks::EmptyCallback, nullptr);
+    accessor_.reset(nullptr);
+
+    // create Table B
+    txn_ = txn_manager_->BeginTransaction();
+    accessor_ = catalog_->GetAccessor(txn_, db_oid_);
+
+    // Create the column definition (no OIDs) for CREATE TABLE b(b1 int, B2 varchar)
+    std::vector<catalog::Schema::Column> cols_b;
+    cols_b.emplace_back("b1", type::TypeId::INTEGER, true, int_default);
+    cols_b.emplace_back("b2", type::TypeId::VARCHAR, 20, true, varchar_default);
+
+    auto schema_b = catalog::Schema(cols_b);
+    table_b_oid_ = accessor_->CreateTable(accessor_->GetDefaultNamespace(), "b", schema_b);
+    auto table_b = new storage::SqlTable(&block_store_, schema_b);
+    EXPECT_TRUE(accessor_->SetTablePointer(table_b_oid_, table_b));
+    txn_manager_->Commit(txn_, TestCallbacks::EmptyCallback, nullptr);
+    accessor_.reset(nullptr);
+  }
+
+  void TearDownTables() {
+    catalog_->TearDown();
+    // Run the GC to flush it down to a clean system
+    gc_->PerformGarbageCollection();
+    gc_->PerformGarbageCollection();
+    gc_->PerformGarbageCollection();
+    delete catalog_;
+    delete gc_;
+    delete txn_manager_;
+    delete deferred_action_manager_;
+    delete timestamp_manager_;
+  }
+
+  void SetUp() override {
+    TerrierTest::SetUp();
+    SetUpTables();
+    // prepare for testing
+    txn_ = txn_manager_->BeginTransaction();
+    accessor_ = catalog_->GetAccessor(txn_, db_oid_);
+    binder_ = new binder::BindNodeVisitor(std::move(accessor_), default_database_name_);
+  }
+
+  void TearDown() override {
+    txn_manager_->Commit(txn_, TestCallbacks::EmptyCallback, nullptr);
+    accessor_.reset(nullptr);
+    delete binder_;
+    delete operator_transformer_;
+    delete operator_tree_;
+    TearDownTables();
+    TerrierTest::TearDown();
+  }
+
+  std::string GetInfo(optimizer::OperatorExpression *op) const {
+    std::string info = "{";
+    {
+      info += "\"Op\":";
+      info += "\"" + op->GetOp().GetName() + "\",";
+      auto children = op->GetChildren();
+      if (!children.empty()) {
+        info += "\"Children\":[";
+        {
+          bool is_first = true;
+          for (const auto &child : children) {
+            if (is_first) {
+              is_first = false;
+            } else {
+              info += ",";
+            }
+            info += GetInfo(child);
+          }
+        }
+        info += "]";
+      }
+    }
+    info += '}';
+    return info;
+  }
+};
```

Comment:
Rename this method to `GenerateOperatorAudit` to make it clear that this method is used for testing purposes. I initially thought that you were using it for human-readable output.